### ISSUE TITLE
NPE When attempting to read from database

### ISSUE THREAD

**Issue by xiphirx** [amplitude/Amplitude-Android#131]
We're seeing the following crash on Fairphone FP2 devices running Android 5.1
We're using sdk version 2.13.2

```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'android.database.Cursor android.database.sqlite.SQLiteDatabase.query(java.lang.String, java.lang.String[], java.lang.String, java.lang.String[], java.lang.String, java.lang.String, java.lang.String, java.lang.String)' on a null object reference
       at com.amplitude.api.DatabaseHelper.queryDb(DatabaseHelper.java:395)
       at com.amplitude.api.DatabaseHelper.getValueFromTable(DatabaseHelper.java:209)
       at com.amplitude.api.DatabaseHelper.getValue(DatabaseHelper.java:197)
       at com.amplitude.api.AmplitudeClient.upgradeSharedPrefsToDB(AmplitudeClient.java:2098)
       at com.amplitude.api.AmplitudeClient.upgradeSharedPrefsToDB(AmplitudeClient.java:2081)
       at com.amplitude.api.AmplitudeClient$1.run(AmplitudeClient.java:241)
       at android.os.Handler.handleCallback(Handler.java:739)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:135)
       at android.os.HandlerThread.run(HandlerThread.java:61)```

**Comment by djih** [comment]
@xiphirx thanks for reporting. Is this only occuring on Fairphone FP2? How often? I wonder if there's something about that particular model.

**Comment by xiphirx** [comment]
It is, http://crashes.to/s/39010695732

**Comment by djih** [comment]
@xiphirx: quick update I have released v2.13.3 with a patch: https://github.com/amplitude/Amplitude-Android/releases/tag/v2.13.3. It should show up in Maven Central within an hour or so.
